### PR TITLE
ECDSA support

### PR DESCRIPTION
**Pull request by flowher**

### CODE REVIEW

**Comment by flowher** [comment]
Please review and merge

**Comment by ronaldtse** [comment]
Thanks @flowher ! @dewyatt and @randombit would you mind reviewing this?

**Comment by flowher** [comment]
There is a small missing part in this implementation. In crypto.c I'm checking if (hash_size < 32). In fact hash size depends on curve that's used. I'll try to fix it today

File: src/lib/crypto.c [hunk 1]
```diff
@@ -198,109 +199,115 @@ pgp_elgamal_encrypt_mpi(const uint8_t *          encoded_m_buf,
...
+          (alg_params == 256) ?
+            PGP_CURVE_NIST_P_256 :
+            (alg_params == 384) ? PGP_CURVE_NIST_P_384 : PGP_CURVE_NIST_P_521;
+
+        pgp_hash_t   tmp;
+        const size_t hash_size =
+          pgp_hash_create(&tmp, seckey->hash_alg) ? pgp_hash_output_length(&tmp) : 0;
+
+        if (hash_size < 32) {
+            RNP_LOG("Hash output length to small (256 required minimum)");
```
Comment (flowher) [inline src/lib/crypto.c]: That's wrong. 
Also check which hash is used when signing

File: src/lib/rnp.c [hunk 2]
```diff
@@ -1240,6 +1240,8 @@ rnp_import_key(rnp_t *rnp, char *f)
     return rnp_key_store_list(io, rnp->pubring, 0);
 }
 
+#define ID_OFFSET 38
```
Comment (randombit) [inline src/lib/rnp.c]: Is this used?

Comment (flowher) [inline src/lib/rnp.c]: Nop. I'll remove it

**Comment by flowher** [comment]
#18

**Comment by randombit** [comment]
LGTM.

I think some more of the code paths for EdDSA and ECDSA can be further merged (like in reading public keys in `packet-parse.c`, and EdDSA OID handling can change to use the `find_curve_by_OID` logic). All minor stuff, I can send a patch for those later.

**Comment by ronaldtse** [comment]
Thanks @randombit for the review (and in advance for the upcoming patch)! I'm going to merge this now.

**Pull request merged**